.github/workflows/ci.yml

```diff
@@ -0,0 +1,105 @@
+name: Build & Test & Clippy
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  cancel_previous_runs:
+    name: Cancel Previous Runs
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+  build:
+    name: Rust check ${{ matrix.check }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        # We need to compile `test-no-std` separately, otherwise we have std leaks from the build-deps.
+        #
+        # `test-no-std` is only meant to check if the `substrate-api-client` compiles to `no_std`, hence
+        # we omit clippy and test there.
+        check: [ cargo build --release --all --exclude test-no-std,
+                 cargo build --release -p test-no-std,
+
+                 cargo clippy --all --exclude test-no-std -- -D warnings,
+                 cargo clippy --examples -- -D warnings,
+
+                 cargo test --all --exclude test-no-std,
+
+                 cargo fmt --all -- --check,
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: init-rust-target
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@v1
+        with:
+          key: ${{ matrix.check }}
+
+      - name: ${{ matrix.check }}
+        run: ${{ matrix.check }}
+
+  examples:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        example: [ example_generic_extrinsic,
+                   example_print_metadata,
+                   example_transfer,
+                   example_get_storage,
+                   example_get_blocks,
+                   example_event_error_details,
+                   example_sudo,
+
+                    # those two fail if they are run against the integritee-node
+#                   example_benchmark_bulk_xt,
```

Comment:
these are important examples. And I would be in favor of using the real node-template for CI. A dependency on integritee-node will haunt us sooner or later

Reply:
We now run all examples against the substrate-node-template from the master branch.